### PR TITLE
Avoid numpy warning for bool labels. 

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description
Probably needs a test???

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/feature/peak.py [hunk 1]
```diff
@@ -127,6 +127,8 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     # operating on each label separately
     if labels is not None:
         label_values = np.unique(labels)
+        if np.issubdtype(label_values.dtype, np.bool):
+            label_values = label_values.astype(np.int8)
```
Comment (jni) [inline skimage/feature/peak.py]: I wonder if bool dtype here means they expect a labeling... Is this related to a bug report or failing/warning test?

Comment (hmaarrfk) [reply skimage/feature/peak.py]: This generates a warning. They implement diff using the subtraction operator so they claim that subtracting book types is no good.

Comment (jni) [reply skimage/feature/peak.py]: Sure but do we know of cases where bool labels have been passed here? it might be safer to just raise on bool

Comment (hmaarrfk) [reply skimage/feature/peak.py]: It was in a doc test. A binary image was constructed and passed in as a label

Comment (jni) [reply skimage/feature/peak.py]: I imagine it's probably used that way by mistake...

Comment (hmaarrfk) [reply skimage/feature/peak.py]: Hmm. Ok I'll find the reference and let you decide.

**Comment by hmaarrfk** [comment]
I can't find the warning I was trying to avoid before. But if somebody does find it, the yshould probably use `scipy.ndimage.label` to label the boolean image first.

**Pull request closed without merging**